sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuSortExec.scala

```diff
@@ -421,29 +422,18 @@ case class GpuOutOfCoreSortIterator(
     while (!pending.isEmpty && sortedSize < targetSize) {
       // Keep going until we have enough data to return
       var bytesLeftToFetch = targetSize
-      val mergedBatch = withResource(ArrayBuffer[SpillableColumnarBatch]()) { pendingSort =>
+      val pendingSort = new util.ArrayDeque[SpillableColumnarBatch]()
```

Comment:
https://www.scala-lang.org/files/archive/api/2.12.15/scala/collection/mutable/ArrayStack.html ?

Reply:
I believe we should stay with the java one. `ArrayStack` was removed in scala 2.13.x, so I'd rather not introduce code that we'll need to remove if we start to build for scala 2.13.

Reply:
Not a blocker but we won't have to change it for 2.13. Scala Maven Plugin can cross-compile it to 2.13 without a code change at the expense of a deprecation warning.
```
warning: value ArrayStack in package mutable is deprecated (since 2.13.0): Use Stack instead of ArrayStack; it now uses an array-based implementation
```